core/templates/pages/topic-editor-page/services/topic-editor-state.service.spec.ts

```diff
@@ -16,191 +16,124 @@
-import { importAllAngularServices } from 'tests/unit-test-utils';
-import { TestBed } from '@angular/core/testing';
 import { TopicUpdateService } from 'domain/topic/topic-update.service';
-// ^^^ This block is to be removed.
-
-import { TranslatorProviderForTests } from 'tests/test.extras';
+import { TopicObjectFactory } from 'domain/topic/TopicObjectFactory.ts';
+import { TopicEditorStateService } from 'pages/topic-editor-page/services/topic-editor-state.service.ts';
 import { Subscription } from 'rxjs';
-
-require('domain/topic/TopicObjectFactory.ts');
-require('domain/topic/topic-update.service.ts');
-require('pages/topic-editor-page/services/topic-editor-state.service.ts');
-
-describe('Topic editor state service', function() {
-  var TopicEditorStateService = null;
-  var TopicObjectFactory = null;
-  var subtopicPageObjectFactory = null;
-  var topicUpdateService = null;
-  var fakeEditableTopicBackendApiService = null;
-  var fakeTopicRightsBackendApiService = null;
-  var secondSubtopicPageObject = null;
-  var secondBackendTopicObject = null;
-  var secondTopicRightsObject = null;
-  var $rootScope = null;
-  var $q = null;
-
-  var testSubscriptions = null;
-  var subtopicPageLoadedSpy = null;
+import { EditableTopicBackendApiService } from 'domain/topic/editable-topic-backend-api.service';
+import { TopicRightsBackendApiService } from 'domain/topic/topic-rights-backend-api.service';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+
+
+describe('Topic editor state service', () => {
+  let topicEditorStateService : TopicEditorStateService;
+  let topicObjectFactory : TopicObjectFactory;
+  let subtopicPageObjectFactory : SubtopicPageObjectFactory;
+  let topicUpdateService :TopicUpdateService;
+  let secondBackendTopicObject = null;
+  let secondTopicRightsObject = null;
+  let mockEditableTopicBackendApiService = null;
+  let mockTopicRightsBackendApiService = null;
+  let subtopicPageObject:SubtopicPageBackendDict = null;
+  let secondSubtopicPageObject: SubtopicPageBackendDict = null;
+
+  let testSubscriptions = null;
+  let subtopicPageLoadedSpy = null;
 
   const topicInitializedSpy = jasmine.createSpy('topicInitialized');
   const topicReinitializedSpy = jasmine.createSpy('topicReinitialized');
 
-  var FakeEditableTopicBackendApiService = function() {
-    var self = {
-      newBackendSubtopicPageObject: null,
-      newBackendTopicObject: null,
-      backendStorySummariesObject: null,
-      failure: null,
-      fetchTopic: null,
-      fetchSubtopicPage: null,
-      updateTopic: null,
-      fetchStories: null
-    };
+  class MockEditableTopicBackendApiService {
+    newBackendSubtopicPageObject= {};
+    newBackendTopicObject = {};
+    backendStorySummariesObject = [];
+    failure: null;
 
-    var _fetchOrUpdateTopic = function() {
-      return $q(function(resolve, reject) {
-        if (!self.failure) {
-          resolve(self.newBackendTopicObject);
+    fetchTopic():Promise<unknown> {
```

Comment:
I think the unknown type is misused here. Can you try and find the actual type for this. Looks like a TopicBackendDict. Do this similarly for other functions as well.

Reply:
Done